docs/content/Checkbox.md

```diff
@@ -0,0 +1,127 @@
+)
+}
+
+<Note variant="warning">
+Checkbox components should always be accompanied by a corresponding label to improve support for assistive technologies.
+</Note>
+
+## Indeterminate example
+
+An `indeterminate` checkbox state should be used if the input value is neither true nor false. This can be useful in situations where you are required to display an incomplete state, or one that is dependent on other input selections to determine a value.
+
+```jsx live
+<>
+  <Box as="form" p={3} pt={0} pb={1} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id="indeterminate-checkbox" onChange={() => {}} indeterminate={true} />
+    <Text as="label" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}} htmlFor="controlled-checkbox">
+      <Text sx={{display: 'block'}}>Default checkbox</Text>
+    </Text>
+  </Box>
+  <Box key={`sub-checkbox-0`} as="form" p={1} pl={6} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id={`sub-checkbox-0`} checked={true} onChange={() => {}} />
+    <Text as="label" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}} htmlFor={`sub-checkbox-0`}>
+      <Text sx={{display: 'block'}}>Checkbox 1</Text>
+    </Text>
+  </Box>
+  <Box key={`sub-checkbox-1`} as="form" p={1} pl={6} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id={`sub-checkbox-1`} checked={false} onChange={() => {}} />
+    <Text as="label" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}} htmlFor={`sub-checkbox-1`}>
+      <Text sx={{display: 'block'}}>Checkbox 2</Text>
+    </Text>
+  </Box>
+  <Box key={`sub-checkbox-2`} as="form" p={1} pl={6} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id={`sub-checkbox-2`} checked={false} onChange={() => {}} />
+    <Text as="label" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}} htmlFor={`sub-checkbox-2`}>
+      <Text sx={{display: 'block'}}>Checkbox 3</Text>
+    </Text>
+  </Box>
+  <Box key={`sub-checkbox-3`} as="form" p={1} pl={6} sx={{display: 'flex', alignItems: 'center'}}>
+    <Checkbox id={`sub-checkbox-3`} checked={false} onChange={() => {}} />
+    <Text as="label" sx={{fontSize: 2, fontWeight: 'bold', marginLeft: 1}} htmlFor={`sub-checkbox-3`}>
+      <Text sx={{display: 'block'}}>Checkbox 4</Text>
+    </Text>
+  </Box>
+</>
+```
+
+## Component props
+
+Native `<input>` attributes are forwarded to the underlying React `input` component and are not listed below.
+
+| Name           | Type    |  Default  | Description                                                                                                                                             |
+| :------------- | :------ | :-------: | :------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| checked        | Boolean | undefined | Optional. Modifies true/false value of the native checkbox                                                                                              |
```

Comment:
Should we document the `onChange` prop? That seems like an essential piece of the "controlled" mode

Reply:
I've added it for convenience. Can you verify it's okay please @colebemis?

Reply:
Looks great 👍